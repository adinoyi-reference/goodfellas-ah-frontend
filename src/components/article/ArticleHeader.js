import React, { Component } from 'react';
import parser from 'react-html-parser';
import { userPlaceholderImage } from '../../mixin';

class ArticleHeader extends Component {
  followButton = () => {
    const { userId, article } = this.props;
    if (userId !== article.authorId) {
      return (<button type="button">Follow</button>);
    }
  }

  render() {
    const { article } = this.props;
    const { read_time, user, title } = article;
    const number = read_time.split(' ')[0];
    const minute = read_time.split(' ')[1];
    return (
      <div className="article-header">
        <div className="author-icon">
          <img src={user.profile.image || userPlaceholderImage} className="author-image" alt="" />
          <div className="article-title">{parser(title)}</div>
        </div>
        <div className="author-name">
          <div className="read-time">
            <div className="number">{number}</div>
            <div className="minutes">{minute}</div>
          </div>
          <div className="name">{`${user.firstname} ${user.lastname}`}</div>
          {this.followButton()}
        </div>
      </div>
    );
  }
}

export default ArticleHeader;
