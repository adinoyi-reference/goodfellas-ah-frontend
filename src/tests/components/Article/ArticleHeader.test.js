import React from 'react';
import { shallow } from 'enzyme';
import ArticleHeader from '../../../components/article/ArticleHeader';

const setup = () => {
  const props = {
    article: {
      read_time: '1 minutes',
      user: {
        firstname: '',
        lastname: '',
        profile: { image: '' },
      },
      title: 'thhisimage',
    },
  };

  const enzymeWrapper = shallow(<ArticleHeader {...props} />);

  return {
    props,
    enzymeWrapper,
  };
};

describe('components', () => {
  describe('ArticleHeader component', () => {
    it('should render as expected', () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();
    });
  });
});
