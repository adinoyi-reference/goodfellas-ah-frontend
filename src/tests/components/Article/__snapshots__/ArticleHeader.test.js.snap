// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components ArticleHeader component should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleHeader
    article={
      Object {
        "read_time": "1 minutes",
        "title": "thhisimage",
        "user": Object {
          "firstname": "",
          "lastname": "",
          "profile": Object {
            "image": "",
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="author-icon"
        >
          <img
            alt=""
            className="author-image"
            src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1541506955/user-placeholder.png"
          />
          <div
            className="article-title"
          >
            thhisimage
          </div>
        </div>,
        <div
          className="author-name"
        >
          <div
            className="read-time"
          >
            <div
              className="number"
            >
              1
            </div>
            <div
              className="minutes"
            >
              minutes
            </div>
          </div>
          <div
            className="name"
          >
             
          </div>
        </div>,
      ],
      "className": "article-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt=""
              className="author-image"
              src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1541506955/user-placeholder.png"
            />,
            <div
              className="article-title"
            >
              thhisimage
            </div>,
          ],
          "className": "author-icon",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "className": "author-image",
              "src": "https://res.cloudinary.com/drmmqcxkc/image/upload/v1541506955/user-placeholder.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "thhisimage",
              ],
              "className": "article-title",
            },
            "ref": null,
            "rendered": Array [
              "thhisimage",
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="read-time"
            >
              <div
                className="number"
              >
                1
              </div>
              <div
                className="minutes"
              >
                minutes
              </div>
            </div>,
            <div
              className="name"
            >
               
            </div>,
            undefined,
          ],
          "className": "author-name",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="number"
                >
                  1
                </div>,
                <div
                  className="minutes"
                >
                  minutes
                </div>,
              ],
              "className": "read-time",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1",
                  "className": "number",
                },
                "ref": null,
                "rendered": "1",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "minutes",
                  "className": "minutes",
                },
                "ref": null,
                "rendered": "minutes",
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " ",
              "className": "name",
            },
            "ref": null,
            "rendered": " ",
            "type": "div",
          },
          undefined,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="author-icon"
          >
            <img
              alt=""
              className="author-image"
              src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1541506955/user-placeholder.png"
            />
            <div
              className="article-title"
            >
              thhisimage
            </div>
          </div>,
          <div
            className="author-name"
          >
            <div
              className="read-time"
            >
              <div
                className="number"
              >
                1
              </div>
              <div
                className="minutes"
              >
                minutes
              </div>
            </div>
            <div
              className="name"
            >
               
            </div>
          </div>,
        ],
        "className": "article-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt=""
                className="author-image"
                src="https://res.cloudinary.com/drmmqcxkc/image/upload/v1541506955/user-placeholder.png"
              />,
              <div
                className="article-title"
              >
                thhisimage
              </div>,
            ],
            "className": "author-icon",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "className": "author-image",
                "src": "https://res.cloudinary.com/drmmqcxkc/image/upload/v1541506955/user-placeholder.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "thhisimage",
                ],
                "className": "article-title",
              },
              "ref": null,
              "rendered": Array [
                "thhisimage",
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="read-time"
              >
                <div
                  className="number"
                >
                  1
                </div>
                <div
                  className="minutes"
                >
                  minutes
                </div>
              </div>,
              <div
                className="name"
              >
                 
              </div>,
              undefined,
            ],
            "className": "author-name",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="number"
                  >
                    1
                  </div>,
                  <div
                    className="minutes"
                  >
                    minutes
                  </div>,
                ],
                "className": "read-time",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1",
                    "className": "number",
                  },
                  "ref": null,
                  "rendered": "1",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "minutes",
                    "className": "minutes",
                  },
                  "ref": null,
                  "rendered": "minutes",
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " ",
                "className": "name",
              },
              "ref": null,
              "rendered": " ",
              "type": "div",
            },
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
